scipy/spatial/meson.build

```diff
@@ -0,0 +1,112 @@
+_spatial_pxd = custom_target('_spatial_pxd',
+  output : ['qhull.pxd', 'setlist.pxd'],
+  input : ['qhull.pxd', 'setlist.pxd'],
+  command : ['cp', '@INPUT@', '-t', '@OUTDIR@'],
+)
+
+qhull_src = ['qhull_src/src/poly_r.c',
+'qhull_src/src/stat_r.c',
+'qhull_src/src/merge_r.c',
+'qhull_src/src/userprintf_r.c',
+'qhull_src/src/userprintf_rbox_r.c',
+'qhull_src/src/usermem_r.c',
+'qhull_src/src/geom_r.c',
+'qhull_src/src/libqhull_r.c',
+'qhull_src/src/user_r.c',
+'qhull_src/src/random_r.c',
+'qhull_src/src/geom2_r.c',
+'qhull_src/src/mem_r.c',
+'qhull_src/src/global_r.c',
+'qhull_src/src/poly2_r.c',
+'qhull_src/src/io_r.c',
+'qhull_src/src/rboxlib_r.c',
+'qhull_src/src/qset_r.c']
+
+qhull = py3.extension_module('qhull',
+  ['qhull.pyx', 'qhull_misc.h', 'qhull_misc.c'] + qhull_src,
+  c_args: '-Wno-cpp',
+  include_directories: [incdir_numpy, '../_lib', '../_build_utils/src', 'qhull_src/src'],
+  dependencies: [py3_dep, lapack],
+  install : true,
+  subdir : 'scipy/spatial')
+
+ckdtree_src = ['ckdtree/src/query.cxx',
+  'ckdtree/src/build.cxx',
+  'ckdtree/src/query_pairs.cxx',
+  'ckdtree/src/count_neighbors.cxx',
+  'ckdtree/src/query_ball_point.cxx',
+  'ckdtree/src/query_ball_tree.cxx',
+  'ckdtree/src/sparse_distances.cxx']
+
+ckdtree_cpp = custom_target('ckdtree_cpp',
+  output : 'ckdtree.cpp',
+  input : 'ckdtree.pyx',
+  command : [cython, '--cplus', '-3', '--fast-fail', '@INPUT@', '-o', '@OUTPUT@']
+)
+
+ckdtree = py3.extension_module('ckdtree',
+  ckdtree_src + [ckdtree_cpp],
+  cpp_args: '-Wno-cpp', # TODO: replace with numpy_nodepr_api
+  include_directories: [incdir_numpy, '../_lib', '../_build_utils/src', 'ckdtree/src'],
+  dependencies: [py3_dep],
+  install : true,
+  subdir : 'scipy/spatial')
```

Comment:
This extension and `_distance_pybind` use the `pre_build_hook` in `setup.py where that adds compiler flags. Can you add the same block here as in `scipy/_lib/_uarray/meson.build`?

```
# FIXME: /EHsc is the default already, and for hidden use gnu_symbol_visibility
# (see fft/_pocketfft)
compiler = meson.get_compiler('cpp')
if compiler.get_id() == 'msvc'
  cpp_args = '/EHsc'
elif compiler.has_argument('-fvisibility=hidden')
  cpp_args = '-fvisibility=hidden'
endif
```